src/main/java/org/gradle/profiler/buildops/BuildOperationInstrumentation.java

```diff
@@ -87,44 +87,54 @@ public Optional<Duration> getTimeToTaskExecution() {
         }
         try {
             try (Stream<String> lines = Files.lines(configurationTimeDataFile.toPath(), StandardCharsets.UTF_8)) {
-                return lines
-                    .reduce((first, second) -> second)
-                    .map(line -> Duration.ofMillis(Long.parseLong(line)));
+                Optional<String> lastLine = lines.reduce((first, second) -> second);
+                return lastLine.map(line -> readExecutionDataLine(line).getTotalDuration());
```

Comment:
Any reason to not use the chaining notation?

Reply:
Compared to other places where we parse the files, it's non-trivial that here we decide to ignore the whole file apart from the last line. IMO like this the intent is much more clear, than a sneaky `reduce` call.